pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -136,7 +136,7 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		if err != nil {
 			return fmt.Errorf("failed to get namespace: %w", err)
 		}
-		if features.Enabled(features.ManagedJobsNamespaceSelector) && !w.managedJobsNamespaceSelector.Matches(labels.Set(ns.GetLabels())) {
+		if features.Enabled(features.ManagedJobsNamespaceSelector) && w.managedJobsNamespaceSelector != nil && !w.managedJobsNamespaceSelector.Matches(labels.Set(ns.GetLabels())) {
```

Comment:
Adding this because I discovered a bug in our unit tests.

If ManagedJobsNamespaceSelector is enabled but there is no managedJobsNamespaceSelector then we hit a nil pointer exception.

I think we cover this via validation but webhooks don't go through that path. So adding a check for nil here.